### PR TITLE
Add Databricks and Vercel token detection rules

### PR DESCRIPTION
**Pull request by azu**
## Summary
This PR adds two new secretlint rules for detecting sensitive tokens from Databricks and Vercel platforms, along with test snapshots and integration into the canary preset.

## Key Changes

- **New Databricks Rule** (`@secretlint/secretlint-rule-databricks`)
  - Detects Databricks personal access tokens (PAT) with pattern: `dapi` prefix + 32 hex characters + optional `-<digit>` suffix
  - Implements case-insensitive matching for hex characters
  - Includes word boundary checks to avoid false positives in variable names
  - Comprehensive test cases covering valid and invalid token formats

- **New Vercel Rule** (`@secretlint/secretlint-rule-vercel`)
  - Detects multiple Vercel token types:
    - Personal access tokens (`vcp_*`)
    - Integration tokens (`vci_*`)
    - App access tokens (`vca_*`)
    - App refresh tokens (`vcr_*`)
    - AI Gateway API keys (`vck_*`)
  - Test snapshots validate detection of all token types

- **Integration Updates**
  - Added both rules to the canary preset (`@secretlint/secretlint-rule-preset-canary`)
  - Updated preset package.json with new rule dependencies
  - Added test snapshots for preset-level validation

- **Test Infrastructure**
  - Created snapshot tests for both new rules
  - Added test cases for edge cases (word boundaries, placeholder patterns)
  - Updated database connection string tests to handle bash variable patterns

## Implementation Details

- Both rules use Unicode-aware regex patterns with word boundary checks (`\p{L}`)
- Rules support configurable allowlist via `allows` option for pattern matching
- Comprehensive documentation and README files included for each rule
- Follows existing secretlint rule structure and conventions

https://claude.ai/code/session_01K8ikpZTY86Rjdfp3zZEHTt

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -336,6 +336,7 @@ Secretlint rules has been implemented as separated modules.
 - [@secretlint/secretlint-rule-linear](./packages/%40secretlint/secretlint-rule-linear)
 - [@secretlint/secretlint-rule-1password](./packages/%40secretlint/secretlint-rule-1password)
 - [@secretlint/secretlint-rule-database-connection-string](./packages/%40secretlint/secretlint-rule-database-connection-string)
+- [@secretlint/secretlint-rule-databricks](./packages/%40secretlint/secretlint-rule-databricks)
 - [@secretlint/secretlint-rule-secp256k1-privatekey](./packages/@secretlint/secretlint-rule-secp256k1-privatekey)
```
[the remark this answers is not in the document]
Comment (azu) [reply README.md:342]: Fixed in [docs(secretlint-rule-databricks): fix fixture comment and list Vercel rule](https://github.com/secretlint/secretlint/pull/1478/commits/7db3d15).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@secretlint/secretlint-rule-preset-canary/test/snapshots/secretlint-rule-databricks/ok.valid/input.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# These should NOT be detected as Databricks tokens
+
+# Too short (only 4 hex chars after dapi)
+dapi1234
+
+# Not a hex body (contains uppercase letters)
```
[the remark this answers is not in the document]
Comment (azu) [reply packages/@secretlint/secretlint-rule-preset-canary/test/snapshots/secretlint-rule-databricks/ok.valid/input.txt]: Fixed in [docs(secretlint-rule-databricks): fix fixture comment and list Vercel rule](https://github.com/secretlint/secretlint/pull/1478/commits/7db3d15).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/@secretlint/secretlint-rule-databricks/test/snapshots/ok.valid/input.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# These should NOT be detected as Databricks tokens
+
+# Too short (only 4 hex chars after dapi)
+dapi1234
+
+# Not a hex body (contains uppercase letters)
```
[the remark this answers is not in the document]
Comment (azu) [reply packages/@secretlint/secretlint-rule-databricks/test/snapshots/ok.valid/input.txt]: Fixed in [docs(secretlint-rule-databricks): fix fixture comment and list Vercel rule](https://github.com/secretlint/secretlint/pull/1478/commits/7db3d15).

---
_Generated by [Claude Code](https://claude.ai/code)_
